roles/network_plugin/macvlan/meta/main.yml

```diff
@@ -1,3 +1,6 @@
 ---
 dependencies:
   - role: network_plugin/cni
+  - role: container-engine/cri-o
+  - role: container-engine/docker
+  - role: container-engine/containerd
```

Comment:
Failed
```
TASK [container-engine/containerd : containerd | Remove any package manager controlled containerd package] ***
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/container-engine/containerd/tasks/main.yml:8
Thursday 16 June 2022  18:40:35 +0000 (0:00:00.123)       0:10:39.463 ********* 
fatal: [instance-2]: FAILED! => {"changed": false, "msg": "'apt-get remove 'containerd.io'' failed: E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.\n", "rc": 100, "stderr": "E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.\n", "stderr_lines": ["E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nSome packages could not be installed. This may mean that you have\nrequested an impossible situation or if you are using the unstable\ndistribution that some required packages have not yet been created\nor been moved out of Incoming.\nThe following information may help to resolve the situation:\n\nThe following packages have unmet dependencies:\n docker-ce : Depends: containerd.io (>= 1.4.1) but it is not going to be installed\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "Some packages could not be installed. This may mean that you have", "requested an impossible situation or if you are using the unstable", "distribution that some required packages have not yet been created", "or been moved out of Incoming.", "The following information may help to resolve the situation:", "", "The following packages have unmet dependencies:", " docker-ce : Depends: containerd.io (>= 1.4.1) but it is not going to be installed"]}
fatal: [instance-1]: FAILED! => {"changed": false, "msg": "'apt-get remove 'containerd.io'' failed: E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.\n", "rc": 100, "stderr": "E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.\n", "stderr_lines": ["E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nSome packages could not be installed. This may mean that you have\nrequested an impossible situation or if you are using the unstable\ndistribution that some required packages have not yet been created\nor been moved out of Incoming.\nThe following information may help to resolve the situation:\n\nThe following packages have unmet dependencies:\n docker-ce : Depends: containerd.io (>= 1.4.1) but it is not going to be installed\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "Some packages could not be installed. This may mean that you have", "requested an impossible situation or if you are using the unstable", "distribution that some required packages have not yet been created", "or been moved out of Incoming.", "The following information may help to resolve the situation:", "", "The following packages have unmet dependencies:", " docker-ce : Depends: containerd.io (>= 1.4.1) but it is not going to be installed"]}
```

Reply:
Maybe we should put the handler in another role (with only handlers) and include this role in both role

Reply:
This actually triggers the install of all 3 container runtimes.

What you would need is something like this:

```yaml
handlers:
  - include: container-engine/cri-o/handlers/main.yaml
  - include: container-engine/containerd/handlers/main.yaml
  - include: container-engine/docker/handlers/main.yaml
```

But this approach is not that elegant.

Centralising all handlers in a dedicated role might actually be the best option, with a bit more refactoring than was originally intended for this small fix but might yield more elegant approaches in future code.

Reply:
Thanks for advices @floryut @cristicalin 
I have time to restart solving this issue.
Let me try the above way.